pkg/steps/template.go

```diff
@@ -39,6 +39,7 @@ const (
 	RefsOrgLabel    = "ci.openshift.io/refs.org"
 	RefsRepoLabel   = "ci.openshift.io/refs.repo"
 	RefsBranchLabel = "ci.openshift.io/refs.branch"
+	finalizer       = "ci.openshift.io/read-final-status"
```

Comment:
Why did we need this? @alvaroaleman might want to reivew, had a lot of experience with gotchas and finalizers recently

Reply:
Motivation is [here][1] and in the commit message.  Review from folks with more finalizer experience would be much appreciated.

[1]: https://github.com/openshift/ci-tools/pull/1257#issuecomment-704683394

Reply:
I've spun #1277 out with just the finalizer bit, hanging it on the existing `Delete` from `podStep.run`.

Reply:
Obsolete with [the pivot to Vadim's approach][1].

[1]: https://github.com/openshift/ci-tools/pull/1257#issuecomment-705131035